lib/questions.rb

```diff
@@ -1,226 +1,292 @@
+  # e.g. [1, 2, 3, 4, 5, 6] becomes [4, 5, 6]
+  def all_elements_except_first_3(array)
+    array.drop(3)
+  end
+
+  # add an element to the beginning of an array
+  def add_element_to_beginning_of_array(array, element)
+    arr = array << element
+    arr.sort!
+  end
+
+  # sort an array of words by their last letter, e.g.
+  # ['sky', 'puma', 'maker'] becomes ['puma', 'maker', 'sky']
+  def array_sort_by_last_letter_of_word(array)
+    array.sort! {|a, b| a.reverse <=> b.reverse}
+  end
+
+  # cut strings in half, and return the first half, e.g.
+  # 'banana' becomes 'ban'. If the string is an odd number of letters
+  # round up - so 'apple' becomes 'app'
+  def get_first_half_of_string(string)
+    string[0, (string.length / 2.to_f).ceil]
+  end
+
+  # turn a positive integer into a negative integer. A negative integer
+  # stays negative
+  def make_numbers_negative(number)
+    number < 0 ? number : number - number * 2
+  end
+
+  # turn an array of numbers into two arrays of numbers, one an array of
+  # even numbers, the other an array of odd numbers
+  # even numbers come first
+  # so [1, 2, 3, 4, 5, 6] becomes [[2, 4, 6], [1, 3, 5]]
+  def separate_array_into_even_and_odd_numbers(array)
+    odd, even = [], []
+    array.each {|num| num % 2 == 0 ? even << num : odd << num}
+    array = []
+    array << even
+    array << odd
+  end
+
+  # count the numbers of elements in an element which are palindromes
+  # a palindrome is a word that's the same backwards as forward
+  # e.g. 'bob'. So in the array ['bob', 'radar', 'eat'], there
+  # are 2 palindromes (bob and radar), so the method should return 2
+  def number_of_elements_that_are_palindromes(array)
+    array.delete_if {|word| word != word.reverse}
+    array.length
+  end
+
+  # return the shortest word in an array
+  def shortest_word_in_array(array)
+    array.min{|a,b| a.size <=> b.size }
+  end
+
+  # return the shortest word in an array
+  def longest_word_in_array(array)
+    array.max { |a, b| a.length <=> b.length }
+  end
+
+  # add up all the numbers in an array, so [1, 3, 5, 6]
+  # returns 15
+  def total_of_array(array)
+    array.inject {|sum, elem| sum + elem}
+  end
+
+  # turn an array into itself repeated twice. So [1, 2, 3]
+  # becomes [1, 2, 3, 1, 2, 3]
+  def double_array(array)
+    arr = array
+    arr.each {|num| array << num}
+  end
+
+  # convert a symbol into a string
+  def turn_symbol_into_string(symbol)
+    symbol.to_s
+  end
+
+  # get the average from an array, rounded to the nearest integer
+  # so [10, 15, 25] should return 17
+  def average_of_array(array)
+    total = 0
+    array.inject{|sum, elem| total = sum + elem}
+    average = (total / array.length.to_f).ceil
+  end
+
+  # get all the elements in an array, up until the first element
+  # which is greater than five. e.g.
+  # [1, 3, 5, 4, 1, 2, 6, 2, 1, 3, 7]
+  # becomes [1, 3, 5, 4, 1, 2]
+  def get_elements_until_greater_than_five(array)
```

Comment:
Unused method argument - `array`. If it's necessary, use `_` or `_array` as an argument name to indicate that it won't be used. You can also write as `get_elements_until_greater_than_five(*)` if you want the method to accept any arguments but don't care about them.